docs/guide/api-plugin.md

```diff
@@ -399,6 +399,10 @@ Vite plugins can also provide hooks that serve Vite-specific purposes. These hoo
   }
   ```
 
+  ::: warning Note
+  This plugin will not run when [`appType`](https://vitejs.dev/config/shared-options.html#apptype) is `"custom"`. HTML middlewares should be handled at the framework level.
+  :::
```

Comment:
I don't think this is true. `appType` doesn't affect if `transformIndexHtml` is called, it's up to the frameworks if they call `server.transformIndexHtml`. If they don't, then the hook will not be called.

If we want to document it here, I think it's more accurate to say that some SSR frameworks may not support this hook, and hence it may not always run for all HTML.

Reply:
Ok I see, more like this?